### PR TITLE
Inform the user if unable to get current location

### PR DESCRIPTION
**Pull request by german-one**
This shows a message box to make the user aware if the location can not be determined. It also re-enables the "Use current location" button to give the user the opportunity to update settings and try again.

### CODE REVIEW

File: src/settings.c [hunk 1]
```diff
@@ -200,7 +205,7 @@ static HRESULT geolocation_callback(void* context, __FIAsyncOperation_1_Windows_
     IFC(coord->lpVtbl->get_Longitude(coord, &longitude));
 
     float coordinates[2] = {(float)latitude, (float)longitude};
-    SendMessageW(context, WM_GEOLOCATION_UPDATED, 0, (LPARAM)&coordinates);
+    SendMessageW(context, WM_GEOLOCATION_UPDATED, 0, (LPARAM)coordinates);
```
Comment (german-one) [inline src/settings.c:208]: This is unrelated and just because I'm in this area anyways. Technically it's been a `float (*)[2]` decayed to `float **`. Luckily it worked when it was casted to `float *` in the dialog callback. Now it's a `float[2]` decayed to `float *`.

**Review by lhecker — APPROVED** [review]
Thank you so much!
